src/core/api/diags_api.cpp

```diff
@@ -42,11 +42,11 @@
 
 using namespace ot;
 
-void otDiagProcessCmdLine(otInstance *aInstance, const char *aString, char *aOutput, size_t aOutputMaxLen)
+otError otDiagProcessCmdLine(otInstance *aInstance, const char *aString, char *aOutput, size_t aOutputMaxLen)
 {
     AssertPointerIsNotNull(aString);
 
-    AsCoreType(aInstance).Get<FactoryDiags::Diags>().ProcessLine(aString, aOutput, aOutputMaxLen);
+    return AsCoreType(aInstance).Get<FactoryDiags::Diags>().ProcessLine(aString, aOutput, aOutputMaxLen);
```

Comment:
What is the reason for changing this to return `otError`?
I think the idea is if there is any error in processing of the cmd, it should be reflected in resulting `aOutput`.

Reply:
Make sense. Thanks.